### PR TITLE
Add Kubeflow Steering Committee election proposal.

### PR DESCRIPTION
**Pull request by james-jwu**
The proposal is adapted from the kNative [Steering Committee election process](https://github.com/knative/community/blob/main/STEERING-COMMITTEE.md#election-procedure).

### CODE REVIEW

**Review by zijianjoy — COMMENTED** [review]
Thank you James! It looks good to me but will wait for other steering committee members to approve.

**Review by theadactyl — APPROVED** [review]

File: proposals/kubeflow-steering-cmmittee-election-proposal.md
Comment (zijianjoy) [reply proposals/kubeflow-steering-cmmittee-election-proposal.md]: Thank you James!

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (thesuperzapper) [inline proposals/kubeflow-steering-committee-election-proposal.md:74]: Let's remove this line, as any such change would require a "constitutional" vote, as described above.

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
It looks good, but I think we should clarify a few things (listed in my comments).

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (thesuperzapper) [inline proposals/kubeflow-steering-committee-election-proposal.md:70]: Before adopting this proposal we MUST replace this TODO.

**Comment by juliusvonkohout** [comment]
I think to prevent spamming we have to require at least one merged PR. No matter the content, documentation also counts.

**Comment by thesuperzapper** [comment]
> I think to prevent spamming we have to require at least one merged PR. No matter the content, documentation also counts.

@juliusvonkohout I think this is also important, as it will ensure that the member has signed the CLA (which is required to have your PR merged).

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (thesuperzapper) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: Also, as raised by @juliusvonkohout in https://github.com/kubeflow/community/pull/645#issuecomment-1697668212, I think putting a minimum of 1 merged PR (even a minor docs update is sufficient) will help prevent spam.

Also, as I raised in https://github.com/kubeflow/community/pull/645#issuecomment-1697886798, this will additionally ensure the member has signed the CLA.

**Review by andreyvelich — COMMENTED** [review]
Thank you for this proposal @james-jwu, I left a few comments.

cc @kubeflow/wg-training-leads @tenzen-y @anencore94 @kuizhiqing @zw0610 for visibility.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (annajung) [reply proposals/kubeflow-steering-committee-election-proposal.md:56]: +1, if not a specific date, I think it's worth mentioning that it should start in Sep/Oct every year.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (johnugeorge) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: Contributions need not be just in dev. @andreyvelich

Comment (andreyvelich) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: I agree, we can explain here what kind of non dev contributions people can make to be eligible to vote.

Comment (tenzen-y) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: Maybe we can refer to CNCF devstats: https://kubeflow.devstats.cncf.io/d/8/dashboards?orgId=1&refresh=15m

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: Yes we should have defined responsibilities of KSC, but I suggest we do it in a different PR.

Comment (thesuperzapper) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: I agree, this proposal is about how to elect the KSC, not how it will run.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:7]: There won't be any disagreement :). Josh and I are planning to take these seats.

Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:7]: The Interim KSC will conduct the elections, with the help of election officers. The initial 2 members are not going to make up the 1st KSC.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:56]: I am open to suggestions. Curious what other people might suggest.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: Borrowed this query from kNative election eligibility: https://kubeflow.devstats.cncf.io/d/9/developer-activity-counts-by-repository-group-table?orgId=1&var-period_name=Last%20year&var-metric=contributions&var-repogroup_name=All&var-country_name=All

Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: Borrowed this query from kNative election eligiblity: https://kubeflow.devstats.cncf.io/d/9/developer-activity-counts-by-repository-group-table?orgId=1&var-period_name=Last%20year&var-metric=contributions&var-repogroup_name=All&var-country_name=All

Currently there are 46 eligible people meeting the 50 cutoff.

I am open to adding the minimum requirement (1 merged PR or document update). It probably can be self attested.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md]: Done.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:77]: TOC=Technical Oversight Committee, SC=Steering Committee. There are both referring to Kubeflow (not CNCF) committees.

We thought about ensuring a single person cannot serve on more than 1 committee, but was worried that we don't have enough people to staff all 3 committees (KSC, TOC and Conformance committee).

/cc @jbottum

Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:77]: Voting eligibility and candidacy eligibility are the same. I don't feel we need more complicated rules here.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:80]: Sounds good. Updated the text.
/cc @jbottum @theadactyl

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (thesuperzapper) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: If we do put a requirement for 1 PR, it can probably be one over all time, rather than in the last year.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (thesuperzapper) [reply proposals/kubeflow-steering-committee-election-proposal.md:74]: To clarify, I am talking about the line `We are committed to an inclusive process and will adapt future eligibility requirements based on community feedback.`

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (andreyvelich) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: @james-jwu Does it make sense to establish KSC responsibilities before election document ?
That might help community to understand how many people should be on KSC.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (juliusvonkohout) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: Yes

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (annajung) [reply proposals/kubeflow-steering-committee-election-proposal.md:56]: I think something similar to [Kuberenetes steering election schedule](https://github.com/kubernetes/steering/blob/main/elections.md#election-schedule-and-operation) should be sufficient. 

To follow the proposed schedule where the new steering committee takes office in December, the election must conclude a week prior, so having the election conclude in November would be ideal to allow more folks to participate, assuming people take time off closer to Dec for the holidays. And, to account for 2-4 weeks of voting as proposed, the election must start in Oct to allow the election to conclude in Nov. So I'm thinking something like this

September
- Election officers appointed
- Election preparation (publish the list of eligible voters, open call for exceptions)

October
- Call for nomination
- Start of election

November
- Conclusion of election
- Results announced within one week after the election concludes

December
- New steering committee members take office

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (andreyvelich) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: @james-jwu How we can deal with contributions that DevStat doesn't count ? E.g. driving meetings, summits, presentations etc. ?

Comment (thesuperzapper) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: @andreyvelich this is already dealt with, see the following line `If a community member has had significant contributions over the past year but is not captured in the dashboard, they will be able to submit an exception form to the Elections Officer(s) who will then review and determine whether this member should be eligible to vote. All exceptions, and the reasons for them, will be recorded in a log that will be available to Steering and the TOC.`

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (thesuperzapper) [reply proposals/kubeflow-steering-committee-election-proposal.md:56]: I agree with @annajung, but do we want to put specific dates?

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: Looking at steering committee charters from:
- [k8s](https://github.com/kubernetes/steering/blob/main/charter.md#direct-responsibilities-of-the-steering-committee): Define and evolve project and group[3](https://github.com/kubernetes/steering/blob/main/charter.md#footnote3) governance structures and policies[4](https://github.com/kubernetes/steering/blob/main/charter.md#footnote4).
- [kNative](https://github.com/knative/community/blob/main/STEERING-COMMITTEE.md#charter): Define and evolve project governance structures and policies, including project roles and how collaborators become members, approvers, leads, and/or administrators. This includes policy for the creation and administration of [working groups](https://github.com/knative/community/blob/main/working-groups/WORKING-GROUPS.md) and committees.
- [Istio](https://github.com/istio/community/blob/master/steering/CHARTER.md#istio-steering-committee):Establishing rules of governance for the Istio project, including creation and ratification of bylaws for the Steering Committee and Technical Oversight Committee.

In these examples, elections rules, steering committee charters, and other governance policies are purview of the steering committee. Any changes will go through public review & feedback, but the steering committee has the authority to make the changes. Appeals from public community members can be made to CNCF. I propose we adopt the same policy.

@theadactyl @jbottum

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: An exception can be submitted. This is documented a little further below:
```
If a community member has had significant contributions over the past year but is not captured in the dashboard, they will be able to submit an exception form to the Elections Officer(s) who will then review and determine whether this member should be eligible to vote. All exceptions, and the reasons for them, will be recorded in a log that will be available to Steering and the TOC.
```

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md]: I agree 2nd option is fairer. I updated the text.

If a company decides to change their representing committee members, they can stand down one of their committee member, that that doesn't guarantee their preferred candidate will get elected. I added another paragraph to clarify.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (jbottum) [reply proposals/kubeflow-steering-committee-election-proposal.md:77]: Yes, contributors should be able to sit on multiple committees.   Perhaps we should just spell out Kubeflow Technical Oversight Committee and Kubeflow Conformance committee, which is an easy change.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (thesuperzapper) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: In reality, I am only trying to prevent changes to the following things without community consensus:

1. The process by which the KSC is elected
2. The powers delegated to the KSC

So perhaps we can specifically mark this election document and the future "KSC Powers" document to require amending by the process I described.

I think it's fine to allow the KSC to manage its own "house rules" so to speak.

Comment (thesuperzapper) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: Thinking about it more, there seems to be two options for how to achieve this "protection":

1. OPTION 1:
      - Proposals to change the "KSC Charter" and "KSC Election Process" can be raised by anyone, and adopted by the KSC, but only on a super-majority vote (with some protection for situations where there are a lot of vacancies)
1. OPTION 2:
      - Proposals to change the "KSC Charter" and "KSC Election Process" can be raised by anyone, and are adopted by a community vote.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (jbottum) [reply proposals/kubeflow-steering-committee-election-proposal.md:80]: I would like to minimize administration and overhead.    A second election sounds like more work.   Perhaps we can ask the candidates if one wants to bow out, then ask if we can flip a coin, if neither work then 2nd election.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (jbottum) [reply proposals/kubeflow-steering-committee-election-proposal.md:74]: I propose to minimize our modifications of the text at this point and keep as is.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (jbottum) [reply proposals/kubeflow-steering-committee-election-proposal.md]: I believe that eleko requires a voters.yaml file.   https://elekto.dev/docs/administration/creating/#the-votersyaml-file.   Perhaps after the 1st version, it become easier as not that many will be add/deleted.   I would like to put the responsibility on the vote to add his/her name to the voters.yaml file and perhaps have the election official approve.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (jbottum) [reply proposals/kubeflow-steering-committee-election-proposal.md:56]: +1 to Anna's outline, it might be easier for the officials to not define specific dates.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (jbottum) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: I propose that those who want to vote will need to add their names to the voters.yaml file in electo, and that the addition / deletion will be approved by the election officer(s).   if the election officials needs justification, they can ask the voter to provide.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (jbottum) [reply proposals/kubeflow-steering-committee-election-proposal.md:77]: let's keep it simple, +1 on same eligibility for voters and candidates.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (jbottum) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: and perhaps we can seed the list with the list that james has already.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (jbottum) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: I propose that the KSC responsibilities / process can be defined after the election that way the KSC members, who are doing the work, will have a voice in what they are responsible for delivering.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (johnugeorge) [reply proposals/kubeflow-steering-committee-election-proposal.md:77]: +1

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (andreyvelich) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: Would it be confusing that the first selected KSCs will establish their responsibilities and what they want to deliver?
I thought, all active Kubeflow members should define what are the KSC liabilities.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (johnugeorge) [reply proposals/kubeflow-steering-committee-election-proposal.md:56]: +1 to Anna

Comment (thesuperzapper) [reply proposals/kubeflow-steering-committee-election-proposal.md:56]: Perhaps we should at least specify December 1st as the date for the new members to take office, for the avoidance of confusion.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (thesuperzapper) [reply proposals/kubeflow-steering-committee-election-proposal.md:68]: @jbottum the [elekto system has a process to apply for an exception to the voting eligibility built in](https://elekto.dev/docs/administration/running/#voter-exceptions), which I think is better than requiring PRs, because it removes the possibility of public shaming if the request is denied.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: I understand the intention of the new proposal, but I think it results in over-governance. The KSC should have the autonomy to be self-governing and make policy changes. I am interested in hearing from others as well.

Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: I can start a proposal on KSC responsibilities. In the interested of making progress, perhaps we don't have to block the election proposal on the KSC responsibilities proposal. The 1st elected KSC will be able to change/amend the KSC charter.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:56]: I added the general timeline proposed by Anna. For the date when new member to take office, I set it to "first week of December" instead of an exact date.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md]: PR updated.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:93]: I think this is reasonable. Here is what I added: "... the election offices may ask the tied candidates to resolve the tie (e.g. one or more candidates could decide to withdraw). If the tie cannot be resolved among the tied candidates, a runoff election will be conducted ..."

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:105]: @zijianjoy also made a comment about place a minimum vote requirement on selecting the "next most preferred candidate", which I think is supported by elekto tool in some way (see [ranking](https://elekto.dev/docs/administration/concluding/#publishing-the-results) mentioned in their doc). On the other hand, I'd like to reduce process if possible. I am interested in hearing from others.
@jbottum @theadactyl @andreyvelich @johnugeorge @annajung

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (andreyvelich) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: Sure, sounds good!

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (andreyvelich) [reply proposals/kubeflow-steering-committee-election-proposal.md:105]: I agree, we can start with simple rule and improve it during our evaluations.

Comment (annajung) [reply proposals/kubeflow-steering-committee-election-proposal.md:105]: Just to accommodate for the possibility of needing to run a special election instead of filling the empty seat with the previous candidate, we can modify the doc to say "... the next most preferred or conduct a special election to fill the empty seat"

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (theadactyl) [reply proposals/kubeflow-steering-committee-election-proposal.md:1]: Agreed, I think the risk of abuse here is quite low, and it's more important (IMO) to focus on support Steering Committee members being able to do their jobs effectively and efficiently.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:105]: @annajung I'd like to be more prescriptive here.

It looks like there is a mix of opinions here. I think that elections can be very expensive in terms of time and effort. Not only we need candidates, but also election officers to volunteer their time and effort to a multi-week/month process. We can probably get an idea of the level of effort needed from our 1st election. For now I am going to keep the process as-in, and we can re-evaluate after our first election.

File: proposals/kubeflow-steering-committee-election-proposal.md
Comment (james-jwu) [reply proposals/kubeflow-steering-committee-election-proposal.md:81]: Added a requirement on minimum 1 merged PR.

**Comment by james-jwu** [comment]
Thanks everyone for all the feedback! I am merging this PR. The elected KSC can iterate on the election process using learnings from our 1st election.

**Comment by zijianjoy** [comment]
/lgtm

Thank you James and everyone for the proposal here!

**Pull request merged**
